sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -222,7 +222,7 @@ private[hive] class HiveClientImpl(
     hiveConf
   }
 
-  private val userName = UserGroupInformation.getCurrentUser.getShortUserName
+  override val userName = UserGroupInformation.getCurrentUser.getShortUserName
```

Comment:
Can we change user name at runtime? If we can then it should be `def` instead of `val`.

Reply:
I'm not 100% sure, but we have been defining userName as `val` from Spark 2.2.0, starting from this commit https://github.com/apache/spark/commit/344f38b04b271b5f3ec2748b34db4e52d54da1bc

Changing it to def is safer anyway (can handle both cases - whether user name is changed or not) so please let me know if we would like to make change.

Reply:
Btw, that would be better to be handled via separate PR if necessary, to isolate two different things "adding test" and "fixing logic".

Reply:
yea, we can investigate it in a separated pr